pyiron/vasp/base.py

```diff
@@ -403,6 +403,16 @@ def collect_output(self):
                 exclude_nodes=self._exclude_nodes_hdf,
             )
 
+    def nbands_convergence_check(self):
+        """
+        Function to check the convergence of nbands in the Vasp calculation
+
+        Returns:
+
+            bool : True if the highest band is unoccupied, False if the highest band is occupied
+        """
+        return np.all(np.isclose(self["output/electronic_structure/occ_matrix"][:,:,-1], 0)) #shape is n_spin x n_kpoints x n_bands
+
     def convergence_check(self):
```

Comment:
Nice implementation. Could you also call `nbands_convergence_check` inside `convergence_check` and then set the status to `not_converged` if the bands aren't converged?

Reply:
Thanks @sudarsan-surendralal  for the suggestion. I will start working on it

Reply:
Actually, can you move this function to `GenericDFTJob`? Since this is useful for both VASP and SPHINX